middle_end/flambda2/simplify/continuation_extra_params_and_args.ml

```diff
@@ -58,35 +58,58 @@ let [@ocamlformat "disable"] print ppf = function
         @[<hov 1>(extra_args@ %a)@]\
         )@]"
       Bound_parameters.print extra_params
-      (Apply_cont_rewrite_id.Map.print Extra_arg.List.print) extra_args
+      (Apply_cont_rewrite_id.Map.print (Or_invalid.print Extra_arg.List.print)) extra_args
 
 let empty = Empty
 
 let is_empty = function Empty -> true | Non_empty _ -> false
 
-let add t ~extra_param ~extra_args =
+let add t ~invalids ~extra_param ~extra_args =
+  if not
```

Comment:
Wouldn't it be simpler in the rest of the code (and avoid this check as well) to take a single `extra_args` of type `Extra_arg.t Or_invalid.t Apply_cont_rewrite_id.Map.t` instead?

Reply:
Actually:
- It would actually complicate the code that computes the extra args for unboxing, which is the only code that uses the new `invalids` path
- it would also complicate all other users (or rather producers) of extra args since it means that we would have to change their code to generate slightly different maps with a bit more allocations